transaction/core/src/tx.rs

```diff
@@ -103,7 +103,8 @@ impl fmt::Debug for TxHash {
 }
 
 /// A CryptoNote-style transaction.
-#[derive(Clone, Eq, PartialEq, Serialize, Deserialize, Message, Digestible)]
+#[derive(Clone, Eq, PartialEq, Serialize, Deserialize, Message, Digestible, Zeroize)]
+#[zeroize(drop)]
```

Comment:
:question: These [docs](https://docs.rs/zeroize/latest/zeroize/) are hard to parse.
> Zeroing memory is hard
> ...
> This crate has custom derive support for the Zeroize trait, gated under the zeroize crate’s zeroize_derive Cargo feature, which automatically calls zeroize() on all members of a struct or tuple struct.

The _automatically calls_ gave me the impression this was during drop.  In their defense they have examples after with comments saying `ZeorizeOnDrop` happens on drop.

Anyway should the `ZeroizeOnDrop` be added to more types in addition to the plain `Zeroize`?

Reply:
Yes, looks like we want both. That's what the Zeroize tests do: https://github.com/RustCrypto/utils/blob/master/zeroize/tests/zeroize_derive.rs

Reply:
> Anyway should the ZeroizeOnDrop be added to more types in addition to the plain Zeroize?

I am not sure right now
I tried to make a list of things that should maybe be zeroized in #2717 

I figured starting with `Tx` was a reasonable place to start, and I wrote a bunch of rationale in the code comments.

It is causing wierd rust compilation issues to make the type Drop when it wasn't previously, so I'd rather not do this to a lot of types in this PR. And the other thing is, zeroizing operations may have nontrivial overhead, so we either have to measure the impact carefully or have a very good reason why it shouldn't impact network performance (ideally both).